apps/agent/contracts/SignatureValidator.sol

```diff
@@ -134,4 +131,24 @@ library SignatureValidator {
 
         return ret;
     }
+
+    // From: https://github.com/Arachnid/solidity-stringutils/blob/master/src/strings.sol
+    function memcpy(uint dest, uint src, uint len) private pure {
+        // Copy word-length chunks while possible
+        for(; len >= 32; len -= 32) {
+            assembly {
+                mstore(dest, mload(src))
+            }
+            dest += 32;
+            src += 32;
+        }
+
+        // Copy remaining bytes
+        uint mask = 256 ** (32 - len) - 1;
+        assembly {
+            let srcpart := and(mload(src), not(mask))
+            let destpart := and(mload(dest), mask)
+            mstore(dest, or(destpart, srcpart))
```

Comment:
Technically this could be optimized since we don't need to make sure the remaining bytes in `dest` are kept, but let's keep it the same as the library :+1:.

Reply:
Good observation, would rather not modify the code